recipes/libcublas/meta.yaml

```diff
@@ -0,0 +1,163 @@
+{% set platform = "windows-x86_64" %}  # [win]
+{% set target_name = "x86_64-linux" %} # [linux64]
+{% set target_name = "ppc64le-linux" %} # [ppc64le]
+{% set target_name = "sbsa-linux" %} # [aarch64]
+{% set target_name = "x64" %} # [win]
+{% set extension = "tar.xz" %}  # [not win]
+{% set extension = "zip" %}  # [win]
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://developer.download.nvidia.com/compute/cuda/redist/{{ name }}/{{ platform }}/{{ name }}-{{ platform }}-{{ version }}-archive.{{ extension }}
+  sha256: 38272cc9184b4a57a6250fc2cd914ffc827f9f8cd5a3e1356bc3c31f843caf8f  # [linux64]
+  sha256: 2d46e735f8e76e2eb7df1b4f41d56aa8d5d2683f99988bac82d6ae33fb6d344b  # [ppc64le]
+  sha256: 96b68f6821684f39f338862639ce74a74c75830bf44dcd17ef372a1cb8c2304d  # [aarch64]
+  sha256: 88f29bec81880844da2eee13f1cdc51e1ccd8b4c60af6df633577810b30f80d0  # [win]
+
+build:
+  number: 0
+  skip: true  # [osx]
+
+outputs:
+  - name: libcublas
+    files:
+      - lib/lib*blas*.so.*                            # [linux]
+      - targets/{{ target_name }}/lib/lib*blas*.so.*  # [linux]
+      - Library\bin\cublas*.dll                       # [win]
+      - Library\bin\nvblas*.dll                       # [win]
+    requirements:
+      build:
+        - {{ compiler("c") }}
+        - {{ compiler("cxx") }}
+        - sysroot_{{ target_platform }} 2.17   # [linux]
+      host:
+         - cuda-version {{ cuda_version }}
+      run:
+         - {{ pin_compatible("cuda-version", max_pin="x.x") }}
+    test:
+      commands:
+        - test -L $PREFIX/lib/libcublas.so.{{ version }}                                          # [linux]
+        - test -L $PREFIX/lib/libcublas.so.{{ version.split(".")[0] }}                            # [linux]
+        - test -L $PREFIX/targets/{{ target_name }}/lib/libcublas.so.{{ version }}                # [linux]
+        - test -f $PREFIX/targets/{{ target_name }}/lib/libcublas.so.{{ version.split(".")[0] }}  # [linux]
+        - if not exist %LIBRARY_BIN%\cublas64_{{ version.split(".")[0] }}.dll exit 1              # [win]
+        - if not exist %LIBRARY_BIN%\cublasLt64_{{ version.split(".")[0] }}.dll exit 1            # [win]
+        - if not exist %LIBRARY_BIN%\nvblas64_{{ version.split(".")[0] }}.dll exit 1              # [win]
+    about:
+      home: https://developer.nvidia.com/cublas
+      license: LicenseRef-NVIDIA-End-User-License-Agreement
+      license_file: LICENSE
+      license_url: https://docs.nvidia.com/cuda/eula/index.html
+      summary: An implementation of BLAS (Basic Linear Algebra Subprograms) on top of the NVIDIA CUDA runtime.
+      description: |
+        The cuBLAS Library provides a GPU-accelerated implementation of the basic linear algebra subroutines (BLAS).
+      doc_url: https://docs.nvidia.com/cuda/cublas/
+
+  - name: libcublas-dev
+    build:
+      run_exports:
+        - {{ pin_subpackage("libcublas", max_pin="x") }}
```

Comment:
Currently our symlinks are written like:
```
libcublas.so -> libcublas.so.12.0.1.189
```
Because we don't pin to an exact version of `libcublas`, this path may not exist. Do we need to define our symlinks in `$PREFIX/targets/...` to point to the major version like this?
```
libcublas.so -> libcublas.so.12 
```

Reply:
Maybe we need an exact pinning on `requirements: run: libcublas` but _not_ the `run_exports` line?

Reply:
Added a suggestion below ( https://github.com/bdice/staged-recipes/pull/1#discussion_r1151335007 ), which I think is relevant here

Reply:
I don’t think that is sufficient? It would make the link resolvable at build time but that isn’t the issue. The issue is that dev packages are not requiring the exact same libcublas version in the run section, and that means it’s possible to have dead symlinks when installing mismatched versions.

Reply:
Could we test it and see? Unsure how the links can be resolved now (given they don't have a destination to point to)

Adding an exact pin to `run` seemed reasonable to me when we discussed previously. Though thought there were reasons we didn't want to do that (like `*-dev` packages used as `run` dependencies of things like RAFT)

Reply:
Ah, right. I think we have to do that change here and let the end users of raft-headers where it is in a host section (cuml, cugraph) pin their own compatible run dependencies.

Reply:
The symlinks don’t have to resolve to be packaged. If installed alongside the same version of libcublas, they point to the correct files. I don’t think we have to list libcublas in host here.

Reply:
Is there some conda build step that messes with symlinks?

Here's what we have in the source:
```
$ ls -l libcublas-linux-x86_64-12.0.1.189-archive/lib/
...
libcublas.so -> libcublas.so.12
libcublas.so.12 -> ./libcublas.so.12.0.1.189
libcublas.so.12.0.1.189
...
```

I built this package locally and near the end of the build log for `libcublas-dev` I see:
```
Making absolute symlink relative (targets/x86_64-linux/lib/libcublas.so -> libcublas.so.12 :-> libcublas.so.12.0.1.189)
Making absolute symlink relative (targets/x86_64-linux/lib/libnvblas.so -> libnvblas.so.12 :-> libnvblas.so.12.0.1.189)
Making absolute symlink relative (targets/x86_64-linux/lib/libcublasLt.so -> libcublasLt.so.12 :-> libcublasLt.so.12.0.1.189)
```
Which still looks fine to me, but I'm not sure what the actual steps here are.

Reply:
I see. We don't want the symlinks rewritten. We want it to point to `libcublas.so.12` and not `libcublas.so.12.0.1.189`. I don't know how to prevent conda from doing this, it appears mandatory: https://github.com/conda/conda-build/blob/8f558b058807a1c4deb5ff00f3c8b6d0c8d94de2/conda_build/post.py#L1348

The exact pinning I added in 59cc417 works around this, because the exact version will be guaranteed to exist too.

Reply:
This may be an issue. Discussing with @robertmaynard, we need to ensure that compiled programs link to `libcublas.so.12` and not the patch version. We're going to add a test for this.

Reply:
test verified that this is okay, but it did find other issues.

Thinking about this more, we could run something like `readelf -a libcublas.so... | grep SONAME` and maybe verify it is `libcublas.so.<CTK_MAJOR>`?

Reply:
> Thinking about this more, we could run something like `readelf -a libcublas.so... | grep SONAME` and maybe verify it is `libcublas.so.<CTK_MAJOR>`?

I think we ought to defer on this and assume the SONAME is correct in the upstream sources. I don't know if we'll have `readelf` available.

Reply:
That works for me